HW1/P5/P5_bfs.py

```diff
@@ -0,0 +1,221 @@
+import pickle
+import string
+import re
+import urllib2
+import itertools
+import math as ma
+import os
+import pyspark
+
+sc = pyspark.SparkContext()
+sc.setLogLevel('WARN')
+
+def link_string_to_KV(s):
+    src, dests = s.split(': ')
+    dests = [int(to) for to in dests.split(' ')]
+    return (int(src), dests)
+
+
+def fun(i):
+    return lambda x: (x[0], i + 1)
+
+
+def copartitioned(RDD1, RDD2):
+    "check if two RDDs are copartitioned"
+    return RDD1.partitioner == RDD2.partitioner
+
+
+def find_distance(node, dist):
+        for i in dist:
+                if i[0] == node:
+                        position = i[1]
+        return position
+
+
+links = sc.textFile('s3://Harvard-CS205/wikipedia/links-simple-sorted.txt', 32)
+page_names = sc.textFile('s3://Harvard-CS205/wikipedia/titles-sorted.txt', 32)
+
+page_names = page_names.zipWithIndex().map(lambda (n, id): (n, id + 1))
+
+Kevin_Bacon = page_names.filter(lambda (K, V): K == 'Kevin_Bacon').collect()
+assert len(Kevin_Bacon) == 1
+Kevin_Bacon = unicode(Kevin_Bacon[0][1])
+
+Harvard_University = page_names.filter(lambda (K, V): K == 'Harvard_University').collect()
+assert len(Harvard_University) == 1
+Harvard_University = unicode(Harvard_University[0][1])
+
+regex = re.compile(r'[%s\s]+' % re.escape(string.punctuation))

```

Comment:
I appreciate the conciseness of using a regex, but it really does make the code a lot harder to understand. I would recommend that if you use a regex, have a very clear comment that shows what the regex does, and perhaps give a few examples.

In other words, I know what this regex does b/c I remember what we are supposed to do around here, but I'm not sure I would remember in a year.